upgrade/crd.yaml

```diff
@@ -1,68 +1,362 @@
+
+# ZFSVolume CRD is autogenerated via `make manifests` command.
+# Do the modification in the code and run the `make manifests` command
+# to generate the CRD definition
+
+---
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.2.8
+  creationTimestamp: null
   name: zfsvolumes.zfs.openebs.io
 spec:
-  group: zfs.openebs.io
-  version: v1alpha1
-  scope: Namespaced
-  names:
-    plural: zfsvolumes
-    singular: zfsvolume
-    kind: ZFSVolume
-    shortNames:
-    - zfsvol
-    - zv
   additionalPrinterColumns:
   - JSONPath: .spec.poolName
-    name: ZPool
     description: ZFS Pool where the volume is created
+    name: ZPool
     type: string
   - JSONPath: .spec.ownerNodeID
-    name: Node
     description: Node where the volume is created
+    name: Node
     type: string
   - JSONPath: .spec.capacity
-    name: Size
     description: Size of the volume
+    name: Size
     type: string
   - JSONPath: .spec.volblocksize
+    description: volblocksize of volume
     name: volblocksize
-    description: volblocksize for the created zvol
     type: string
   - JSONPath: .spec.recordsize
+    description: recordsize of created zfs dataset
     name: recordsize
-    description: recordsize for the created zfs dataset
     type: string
   - JSONPath: .spec.fsType
-    name: Filesystem
     description: filesystem created on the volume
+    name: Filesystem
     type: string
----
+  - JSONPath: .status.creationTime
+    description: Timestamp when the volume has been created.
+    name: CreationTime
+    type: date
+  - JSONPath: .metadata.creationTimestamp
+    name: Age
+    type: date
+  group: zfs.openebs.io
+  names:
+    kind: ZFSVolume
+    listKind: ZFSVolumeList
+    plural: zfsvolumes
+    shortNames:
+    - zfsvol
+    - zv
+    singular: zfsvolume
+  preserveUnknownFields: false
+  scope: Namespaced
+  subresources: {}
+  validation:
+    openAPIV3Schema:
+      description: ZFSVolume represents a ZFS based volume
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          description: VolumeInfo contains the volume related info for all types of
```

Comment:
The description for VolumeInfo could be changed to something like: spec.Properties defines ZFS volume parameters for all modes in which ZFS volumes can be created like - ZFS volume with filesystem, ZFS Volume exposed as zfs or ZFS volume exposed as raw block device. Some of the parameters can be only set during creation time (as specified in the details of the parameter), and a few are editable. In case of Cloned volumes, the parameters are assigned the same values as the source volume.